OpenRA.Mods.Common/ActorInitializer.cs

```diff
@@ -10,77 +10,124 @@
-		readonly int value = 0;
-
-		public CreationActivityDelayInit() { }
-		public CreationActivityDelayInit(int init) { value = init; }
-		public int Value { get { return value; } }
+		public CreationActivityDelayInit(int value)
+			: base(value) { }
 	}
 
-	public class DynamicFacingInit : IActorInit<Func<int>>
+	public class DynamicFacingInit : ValueActorInit<Func<int>>
 	{
-		readonly Func<int> func;
-
-		public DynamicFacingInit(Func<int> func) { this.func = func; }
-		public Func<int> Value { get { return func; } }
+		public DynamicFacingInit(Func<int> value)
+			: base(value) { }
 	}
 
-	public class SubCellInit : IActorInit<SubCell>
+	public class SubCellInit : ValueActorInit<SubCell>
 	{
-		[FieldFromYamlKey]
-		readonly byte value = (byte)SubCell.FullCell;
+		public SubCellInit(SubCell value)
+			: base(value) { }
+	}
 
-		public SubCellInit() { }
-		public SubCellInit(byte init) { value = init; }
-		public SubCellInit(SubCell init) { value = (byte)init; }
-		public SubCell Value { get { return (SubCell)value; } }
+	public class CenterPositionInit : ValueActorInit<WPos>
+	{
+		public CenterPositionInit(WPos value)
+			: base(value) { }
 	}
 
-	public class CenterPositionInit : IActorInit<WPos>
+	// Allows maps / transformations to specify the faction variant of an actor.
+	public class FactionInit : ValueActorInit<string>
 	{
-		[FieldFromYamlKey]
-		readonly WPos value = WPos.Zero;
+		public FactionInit(string value)
+			: base(value) { }
+	}
 
-		public CenterPositionInit() { }
-		public CenterPositionInit(WPos init) { value = init; }
-		public WPos Value { get { return value; } }
+	public class EffectiveOwnerInit : ValueActorInit<Player>
+	{
+		public EffectiveOwnerInit(Player value)
+			: base(value) { }
 	}
 
-	// Allows maps / transformations to specify the faction variant of an actor.
-	public class FactionInit : IActorInit<string>
+	internal class ActorInitLoader : TypeConverter
 	{
-		[FieldFromYamlKey]
-		public readonly string Faction;
+		public override bool CanConvertFrom(ITypeDescriptorContext context, Type sourceType)
+		{
+			return sourceType == typeof(string) || base.CanConvertFrom(context, sourceType);
+		}
 
-		public FactionInit() { }
-		public FactionInit(string faction) { Faction = faction; }
-		public string Value { get { return Faction; } }
+		public override object ConvertFrom(ITypeDescriptorContext context, System.Globalization.CultureInfo culture, object value)
+		{
+			return new ActorInitActorReference(value as string);
+		}
+
+		public override bool CanConvertTo(ITypeDescriptorContext context, Type destinationType)
+		{
+			return destinationType == typeof(string) || base.CanConvertTo(context, destinationType);
+		}
+
+		public override object ConvertTo(ITypeDescriptorContext context, System.Globalization.CultureInfo culture, object value, Type destinationType)
+		{
+			if (destinationType == typeof(string) && value is ActorInitActorReference)
+				return ((ActorInitActorReference)value).InternalName;
```

Comment:
Although a bit more verbose, the `as` convention strikes again:
```suggestion
			if (destinationType == typeof(string))
			{
				var reference = value as ActorInitActorReference;
				if (reference != null)
					return reference.InternalName;
			}
```